docs/internal-documentation/maintenance/ReleaseProcess-create-docker-image.md

```diff
@@ -0,0 +1,195 @@
+Zonemaster-Engine and Zonemaster-CLI are covered here.
+
+
+## 2. Prerequsite
+
+The steps in this documents are assumet to be executed on a computer installed
+as an [Ubuntu Build Environment] computer. It could work on another OS as long
+as the same support is available.
+
+All commands in this instruction are assumed to be executed from the one and the
+same directory. If you run `cd`, then you have to run `cd` back to the start
+directory.
+
+
+## 3. Create Docker images
+
+Clone the three repositories:
+
+```sh
+git clone https://github.com/zonemaster/zonemaster-ldns
+git clone https://github.com/zonemaster/zonemaster-engine
+git clone https://github.com/zonemaster/zonemaster-cli
+```
+
+Check out right branch depending on the use case.
+
+* Check out `develop` branch when creating an image for release testing:
+
+```sh
+git -C zonemaster-ldns checkout origin/develop
+git -C zonemaster-engine checkout origin/develop
+git -C zonemaster-cli checkout origin/develop
+```
+
+* Check out `master` branch when creating an image for Docker Hub at release, or
+when creating an image based on release version:
+
+```sh
+git -C zonemaster-ldns checkout origin/master
+git -C zonemaster-engine checkout origin/master
+git -C zonemaster-cli checkout origin/master
+```
+
+Create `Makefile` in all three repositories
+
+```sh
+(cd zonemaster-ldns; git clean -dfx; perl Makefile.PL)
+```
+```sh
+(cd zonemaster-engine; git clean -dfx; perl Makefile.PL)
+```
+```sh
+(cd zonemaster-cli; git clean -dfx; perl Makefile.PL)
+```
+
+Create an image for each repository. That image will be tagged "local". The
+images must be created in order since there is a dependency on the previous
+image in each step.
+```sh
+make -C zonemaster-ldns all dist docker-build
+```
+```sh
+make -C zonemaster-engine all docker-build
+```
+```sh
+make -C zonemaster-cli all docker-build
+```
+
+For the Zonemaster-CLI image, add a version tag and a tag "latest".
+
+* Add version tag:
+```sh
+make -C zonemaster-cli docker-tag-version
+```
+
+* Add tag "latest":
+```sh
+make -C zonemaster-cli docker-tag-latest
+```
+
+All the created images can now be listed. Also consider doing [sanity checks] to
+verify that all images work. List images:
+
+```sh
+docker images
+```
+
+## 4. Upload images to Docker Hub
+
+To upload an image to the Zonemaster Docker Hub organization you have to have
+a Docker Hub account and the authorization to upload images.
+
+```sh
+docker login -u $DOCKERUSER
+```
+
+The same image is pushed twice with different tags. Verify in the listing
+above that they have the same ID.
+
+* Push latest.
+```sh
+docker push zonemaster/cli:latest
+```
+
+* Set correct version (see listing) and push image with version tag:
+```sh
+docker push zonemaster/cli:v0.0.0
+```
+
+## 5. Image sanity checks
+
+Zonemaster-LDNS:
+
+```sh
+docker run --rm --entrypoint=perl zonemaster/ldns:local -MZonemaster::LDNS -E 'say Zonemaster::LDNS->new("9.9.9.9")->query("zonemaster.net")->string'
```

Comment:
Since the ldns image doesn't have a default entrypoint you can express this more simply like this. The same goes for the engine image.
```sh
docker run --rm zonemaster/ldns:local perl -MZonemaster::LDNS -E 'say Zonemaster::LDNS->new("9.9.9.9")->query("zonemaster.net")->string'
```

Reply:
Updated both.